src/targets/gpu/include/migraphx/gpu/context.hpp

```diff
@@ -126,6 +127,41 @@ struct hip_device
         }
 #endif
 
+#if MIGRAPHX_USE_HIPBLASLT
+        auto get_hipblaslt()
+        {
+            setup();
+            if(hblthandle == nullptr)
+            {
+                hblthandle = create_hipblaslt_handle_ptr(get());
+            }
+            assert(hblthandle.get() != nullptr);
+            return hblthandle.get();
+        }
+
+        auto get_hipblaslt_preference()
+        {
+            setup();
+            if(hbltpreference == nullptr)
+            {
+                hbltpreference = create_hipblaslt_preference_ptr();
+            }
+            assert(hbltpreference.get() != nullptr);
+            return hbltpreference.get();
+        }
+
+        auto get_hipblaslt_workspace()
+        {
+            setup();
+            if(hbltworkspace == nullptr)
+            {
+                hbltworkspace = create_hipblaslt_workspace_ptr();
+            }
+            assert(hbltworkspace.get() != nullptr);
+            return hbltworkspace.get();
+        }
```

Comment:
The workspace should not be stored in the context. Instead during lowering we can add a allocate instruction and then pass it as an argument to the operator. This is the way we handle the workspace in miopen.

Reply:
Working on this...

Reply:
Made the changes to add `allocate_gpu` in lowering.

Reply:
So the change in lowering for `allocate_gpu` would not work. I have removed the change and I will discuss this with Paul when he is back to see what we should do here.

Reply:
What the issue that you ran into?

Reply:
The workspace should be added before the output parameter.

Reply:
Done.